### DISCUSSION TITLE
'z-shell/H-S-MW' plugin wasn't loading

### DISCUSSION THREAD

**Discussion by awarewen** [z-shell/zi#278] [Q&A]
Plugin: ' z-shell/H-S-MW '

-- Expected performance
![](https://user-images.githubusercontent.com/59910950/182902165-a2e82ec4-c4e8-49a7-a83f-036e0a698655.gif)

-- Post-installation performance
![图片](https://github.com/z-shell/zi/assets/53365857/2a138f93-3e87-421b-9c1c-bdfc4294f988)

-- ZI_CONF:
```
zi light-mode for \
  z-shell/z-a-meta-plugins \
  @annexes+ \
  @zsh-users+fast \
  @fuzzy \
  @romkatv \
  skip'F-Sy-H zsh-diff-so-fancy' @z-shell

typeset -gA HSMW_HIGHLIGHT_STYLES
zstyle :plugin:history-search-multi-word reset-prompt-protect 1
HSMW_HIGHLIGHT_STYLES[/home/awarewen/.histfile]="bg=magenta,fg=white,bold"
zstyle ":history-search-multi-word" page-size "8"                      # Number of entries to show (default is $LINES/3)
zstyle ":history-search-multi-word" highlight-color "fg=yellow,bold"   # Color in which to highlight matched, searched text (default bg=17 on 256-color terminals)
zstyle ":plugin:history-search-multi-word" synhl "yes"                 # Whether to perform syntax highlighting (default true)
zstyle ":plugin:history-search-multi-word" active "underline"          # Effect on active history entry. Try: standout, bold, bg=blue (default underline)
zstyle ":plugin:history-search-multi-word" check-paths "yes"           # Whether to check paths for existence and mark with magenta (default true)
zstyle ":plugin:history-search-multi-word" clear-on-cancel "no"        # Whether pressing Ctrl-C or ESC should clear entered query

```

Reading the wiki, this suggests that the plugin "H-S-MW" should work out of the box, is it my configuration that ignores any switch  :question: 
I used "mcfly" in "zsh_shell", but there was a bug that crashed my computer and I was going to switch to "H-S-MW".

I hope that "H-S-MW" can achieve the performance interface in the profile, what do I still need to do? :smiley:

**Comment by awarewen** [comment]
finded this and When I manually load the plugin it works fine

- 'bindkey' info:
![图片](https://github.com/z-shell/zi/assets/53365857/4d02f708-bdbd-4486-b3dc-bc42b085a6a9)
______
![图片](https://github.com/z-shell/zi/assets/53365857/227df704-65ef-4990-ae91-f412b77534ad)



-- After manually loaded 'bindkey' info:
![图片](https://github.com/z-shell/zi/assets/53365857/e49627f9-3ffc-4ba9-91ee-af7aed2c4528)
_____
![图片](https://github.com/z-shell/zi/assets/53365857/3331c4f1-ccf1-4cd2-843c-9237ff9b2e0b)



I didn't figure out why the plugin wasn't loading.  :crying_cat_face:

**Comment by awarewen** [comment]
`zi ice wait lucid; zi light z-shell/H-S-MW` works fine.

**Reply by ss-o** [reply]
Hi @awarewen,

If I understood correctly, the plugin is not loaded when installed as a meta-plugin with the snippet below:

```zsh
zi light-mode for \
  z-shell/z-a-meta-plugins \
  @annexes+ \
  @zsh-users+fast \
  @fuzzy \
  @romkatv \
  skip'F-Sy-H zsh-diff-so-fancy' @z-shell
```

I will check what might be the cause and update you on this :dart:
